Sources/NIO/Bootstrap.swift

```diff
@@ -439,7 +417,19 @@ public final class ClientBootstrap: NIOTCPClientBootstrap {
     /// - parameters:
     ///     - handler: A closure that initializes the provided `Channel`.
     public func channelInitializer(_ handler: @escaping (Channel) -> EventLoopFuture<Void>) -> Self {
-        self.channelInitializer = handler
+        self._channelInitializer = handler
+        return self
+    }
+
+    /// Sets the protocol handlers that will be added to the front of the `ChannelPipeline` right after the
+    /// `channelInitializer` has been called.
+    ///
+    /// Per bootstrap, you can only set the `protocolHandlers` once. Typically, `protocolHandlers` are used for the TLS
+    /// implementation. Most notably, `NIOClientTCPBootstrap`, NIO's "universal bootstrap" abstraction, uses
+    /// `protocolHandlers` to add the requires `ChannelHandler`s for many TLS implementations.
+    public func protocolHandlers(_ handlers: @escaping () -> [ChannelHandler]) -> Self {
+        precondition(self.protocolHandlers == nil, "protocol handlers can only be set once")
```

Comment:
Why do we enforce this for protocol handlers but not for the channel initializer?

Reply:
it's a good q. My rationale was that because the new `NIOTLSProvider`s may use the `protocolHandlers` it may lead to something unexpected if the users would use the `protocolHandlers` for something and then later on call `enableTLS` (which for NIOSSL will set the `protocolHandlers` without the user seeing that)...

I _think_ that's more likely to happen than the _user_ calling `channelInitializer` twice themselves.

CC @Lukasa

Reply:
Seems reasonable to me.

Reply:
Makes sense; thanks for clarifying.